packages/grafana-ui/src/components/Table/TableNG/utils.ts

```diff
@@ -0,0 +1,98 @@
+import { FieldType } from '@grafana/data';
+
+export function getCellHeight(
+  text: string,
+  cellWidth: number, // width of the cell without padding
+  osContext: OffscreenCanvasRenderingContext2D | null,
+  lineHeight: number,
+  defaultRowHeight: number,
+  padding = 0
+) {
+  const PADDING = padding * 2;
+
+  if (osContext !== null && typeof text === 'string') {
+    const words = text.split(/\s/);
+    const lines = [];
+    let currentLine = '';
+
+    // Let's just wrap the lines and see how well the measurement works
+    for (let i = 0; i < words.length; i++) {
+      const currentWord = words[i];
+      // TODO: this method is not accurate
+      let lineWidth = osContext.measureText(currentLine + ' ' + currentWord).width;
+
+      // if line width is less than the cell width, add the word to the current line and continue
+      // else add the current line to the lines array and start a new line with the current word
+      if (lineWidth < cellWidth) {
+        currentLine += ' ' + currentWord;
+      } else {
+        lines.push({
+          width: lineWidth,
+          line: currentLine,
+        });
+
+        currentLine = currentWord;
+      }
+
+      // if we are at the last word, add the current line to the lines array
+      if (i === words.length - 1) {
+        lines.push({
+          width: lineWidth,
+          line: currentLine,
+        });
+      }
+    }
+
+    // TODO: double padding to adjust osContext.measureText() results
+    const height = lines.length * lineHeight + PADDING * 2;
+
+    return height;
+  }
+
+  return defaultRowHeight;
+}
+
+export function getRowHeight(
+  row: Record<string, string>,
+  columnTypes: Record<string, string>,
+  headerCellRefs: React.MutableRefObject<Record<string, HTMLDivElement>>,
+  osContext: OffscreenCanvasRenderingContext2D | null,
+  lineHeight: number,
+  defaultRowHeight: number,
+  padding: number,
+  textWrap: boolean
+): number {
+  if (!textWrap) {
+    return defaultRowHeight;
+  }
+  /**
+   * 0. loop through all cells in row
+   * 1. find text cell in row
+   * 2. find width of text cell
+   * 3. calculate height based on width and text length
+   * 4. return biggest height
+   */
+
+  let biggestHeight = defaultRowHeight;
+
+  for (const key in row) {
+    if (isTextCell(key, columnTypes)) {
+      if (Object.keys(headerCellRefs.current).length === 0 || !headerCellRefs.current[key]) {
```

Comment:
I had to add this check because I had a table where `headerCellRefs.current[key]` was undefined, causing `const cellWidth = headerCellRefs.current[key].offsetWidth;` to crash the panel.